src/jsonUtils/computeNode.js

```diff
@@ -0,0 +1,445 @@
+// @flow
+import * as yoga from 'yoga-layout';
+import type { TreeNode, ViewStyle } from '../types';
+import Context from '../utils/Context';
+import createStringMeasurer from '../utils/createStringMeasurer';
+import hasAnyDefined from '../utils/hasAnyDefined';
+import pick from '../utils/pick';
+import { INHERITABLE_FONT_STYLES } from '../utils/constants';
+
+// flatten all styles (including nested) into one object
+const getStyles = (node: TreeNode): ViewStyle | Object => {
+  let style = node.props.style;
+
+  if (Array.isArray(style)) {
+    const flattened = Array.prototype.concat.apply([], style);
+    const themeFlattened = Array.prototype.concat.apply([], flattened);
+    const objectsOnly = themeFlattened.filter(f => f);
+    style = Object.assign({}, ...objectsOnly);
+  }
+
+  return style;
+};
+
+const isPercent = (value: number | string) => /%/.test(String(value));
+
+const isAuto = (value: number | string) => String(value) === 'auto';
+
+const isNullOrUndefined = (value: any) => value === null || value === undefined;
+
+const computeNode = (node: TreeNode, context: Context) => {
+  const yogaNode = yoga.Node.create();
+  const hasStyle = node.props && node.props.style;
+  const style: ViewStyle | Object = hasStyle ? getStyles(node) : {};
+
+  if (hasStyle) {
+    // http://facebook.github.io/react-native/releases/0.48/docs/layout-props.html
+
+    // Width
+    if (!isNullOrUndefined(style.width)) {
+      if (isPercent(style.width)) {
+        yogaNode.setWidthPercent(style.width);
+      } else if (isAuto(style.width)) {
+        yogaNode.setWidthAuto();
+      } else {
+        yogaNode.setWidth(style.width);
+      }
+    }
+
+    // Height
+    if (!isNullOrUndefined(style.height)) {
+      if (isPercent(style.height)) {
+        yogaNode.setHeightPercent(style.height);
+      } else if (isAuto(style.height)) {
+        yogaNode.setHeightAuto();
+      } else {
+        yogaNode.setHeight(style.height);
+      }
+    }
+
+    // Min-Height
+    if (!isNullOrUndefined(style.minHeight)) {
+      if (isPercent(style.minHeight)) {
+        yogaNode.setMinHeightPercent(style.minHeight);
+      } else {
+        yogaNode.setMinHeight(style.minHeight);
+      }
+    }
+
+    // Min-Width
+    if (!isNullOrUndefined(style.minWidth)) {
+      if (isPercent(style.minWidth)) {
```

Comment:
Apparently this is handled [here](https://github.com/facebook/yoga/blob/master/javascript/sources/entry-common.js#L154-L155)

Reply:
Nice catch! Updated to remove that extra code